### ISSUE TITLE
Compilation error on FreeBSD

### ISSUE THREAD

**Issue by MrDefacto** [slact/nchan#506]
Hi,

Just wanna report:

`--- objs/addon/redis/redis_nodeset.o ---
/usr/ports/www/nginx/work/nchan/src/store/redis/redis_nodeset.c:1926:17: error: no member named 'sockaddr' in 'struct redisContext'
  switch(ctx->c.sockaddr.sa_family) {
         ~~~~~~ ^
1 error generated.
*** [objs/addon/redis/redis_nodeset.o] Error code 1`

Regards,
D

**Comment by slact** [comment]
I'm unable to reproduce this on FreeBSD 12 with clang or gcc. Please post your env info (FreeBSD version, nchan version, compiler, ./configure parameters and CFLAGS).

**Comment by MrDefacto** [comment]
1. System info:
FreeBSD 11.2-STABLE #25 r342927
FreeBSD clang version 6.0.1 (tags/RELEASE_601/final 335540) (based on LLVM 6.0.1)
latest nchan from github master

2. Steps to reproduce:

`
'download latest version of FreeBSD ports using for example svn'

'go to nginx ports'
# cd /usr/ports/www/nginx

'edit file Makefile and add lines:

CONFIGURE_ARGS+= add-dynamic-module=/usr/ports/www/nginx/work/nchan \
                 with-cc-opt="-I ${LOCALBASE}/include/hiredis"
LIB_DEPENDS+=libhiredis.so:databases/hiredis
'

'execute following commands'

# make patch
# cd work
# git clone https://github.com/slact/nchan.git
# cd ..
# make

`

**Comment by slact** [comment]
remove 
```
with-cc-opt="-I ${LOCALBASE}/include/hiredis"
LIB_DEPENDS+=libhiredis.so:databases/hiredis
```
and it should work. Nchan uses a hacked bundled version of hiredis and does not depend on he standard hiredis lib.

**Comment by MrDefacto** [comment]
no. it doesn't help.

`cc -c -fPIC -O2 -pipe  -I/usr/local/include -Wl,-rpath,/usr/local/lib -fstack-protector -fno-strict-aliasing  -I /usr/local/include -I /usr/local/include/hiredis -DNDK_SET_VAR -I src/core  -I src/event  -I src/event/modules  -I src/os/unix  -I /usr/ports/www/nginx/work/nginx_upstream_check_module-9aecf15  -I /usr/ports/www/nginx/work/nchan/src  -I /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/objs  -I objs/addon/ndk  -I /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/src  -I /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/objs  -I objs/addon/ndk  -I objs  -I src/http  -I src/http/modules  -I src/http/v2  -I /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/src  -I src/stream  -o objs/ngx_nchan_module_modules.o  objs/ngx_nchan_module_modules.c
cc: warning: -Wl,-rpath,/usr/local/lib: 'linker' input unused [-Wunused-command-line-argument]
--- objs/addon/src/ndk.o ---
cc -c -fPIC -O2 -pipe  -I/usr/local/include -Wl,-rpath,/usr/local/lib -fstack-protector -fno-strict-aliasing  -I /usr/local/include -I /usr/local/include/hiredis -DNDK_SET_VAR -I src/core  -I src/event  -I src/event/modules  -I src/os/unix  -I /usr/ports/www/nginx/work/nginx_upstream_check_module-9aecf15  -I /usr/ports/www/nginx/work/nchan/src  -I /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/objs  -I objs/addon/ndk  -I /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/src  -I /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/objs  -I objs/addon/ndk  -I objs  -I src/http  -I src/http/modules  -I src/http/v2  -I /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/src  -I src/stream  -o objs/addon/src/ndk.o  /usr/ports/www/nginx/work/ngx_devel_kit-0.3.0/src/ndk.c
cc: warning: -Wl,-rpath,/usr/local/lib: 'linker' input unused [-Wunused-command-line-argument]
--- objs/addon/redis/redis_nodeset.o ---
/usr/ports/www/nginx/work/nchan/src/store/redis/redis_nodeset.c:1926:17: error: no member named 'sockaddr' in 'struct redisContext'
  switch(ctx->c.sockaddr.sa_family) {
         ~~~~~~ ^
1 error generated.
*** [objs/addon/redis/redis_nodeset.o] Error code 1

make[3]: stopped in /usr/ports/www/nginx/work/nginx-1.14.2
1 error

make[3]: stopped in /usr/ports/www/nginx/work/nginx-1.14.2
*** [build] Error code 2

make[2]: stopped in /usr/ports/www/nginx/work/nginx-1.14.2
1 error

make[2]: stopped in /usr/ports/www/nginx/work/nginx-1.14.2
===> Compilation failed unexpectedly.
Try to set MAKE_JOBS_UNSAFE=yes and rebuild before reporting the failure to
the maintainer.
*** Error code 1

Stop.
make[1]: stopped in /usr/ports/www/nginx
*** Error code 1

Stop.
make: stopped in /usr/ports/www/nginx`

**Comment by slact** [comment]
tried it on FreeBSD12, worked fine. try to rm work, make clean, and rebuild from scratch without 
```
with-cc-opt="-I ${LOCALBASE}/include/hiredis"
LIB_DEPENDS+=libhiredis.so:databases/hiredis
```

**Comment by MrDefacto** [comment]
even if i remove those line... but hiredis is installed in system... compilation fails. after 'pkg delete hiredis' compilation works.

it looks that there is some kind of conflict.

**Comment by slact** [comment]
can you post the entire build output?

**Comment by MrDefacto** [comment]
sent to support email

**Comment by slact** [comment]
alright, try the `hiredis-buildfix` branch. Liike like FreeBSD prioritizes system includes over local ones, so the bundled hiredis header was getting clobbered by the shared one.

**Comment by MrDefacto** [comment]
in theory that is how it should be. especially if in system is that same version that is bundled. i've tested hiredis-buildfix and it works 👍 even if hiredis is installed in system it compiles correclty. On FreeBSD 11 also. Thank You! GJ

**Comment by slact** [comment]
Fix is in version 1.2.4.
